src/test/java/com/exasol/adapter/dialects/saphana/util/IntegrationTestSetup.java

```diff
@@ -0,0 +1,189 @@
+    private static final Logger LOGGER = Logger.getLogger(IntegrationTestSetup.class.getName());
+
+    private final HanaContainer<?> hanaContainer;
+    private final ExasolContainer<?> exasolContainer;
+    private final Connection exasolConnection;
+    private final Connection hanaConnection;
+    private final ExasolObjectFactory exasolFactory;
+    private final HanaObjectFactory hanaFactory;
+    private final AdapterScript adapterScript;
+    private final ConnectionDefinition connectionDefinition;
+    private int virtualSchemaCounter = 0;
+    private int hanaSchemaCounter = 0;
+
+    public IntegrationTestSetup(final HanaContainer<?> hana, final ExasolContainer<?> exasol)
+            throws NoDriverFoundException, SQLException {
+        this.hanaContainer = hana;
+        this.exasolContainer = exasol;
+        this.exasolConnection = this.createExasolConnection();
+        this.hanaConnection = DriverManager.getConnection(this.hanaContainer.getJdbcUrl(), hanaContainer.getUsername(),
+                hanaContainer.getPassword());
+        final ExasolObjectConfiguration.Builder builder = ExasolObjectConfiguration.builder();
+        final UdfTestSetup udfTestSetup = new UdfTestSetup(getTestHostIpFromInsideExasol(),
+                this.exasolContainer.getDefaultBucket(), this.exasolConnection);
+        builder.withJvmOptions(udfTestSetup.getJvmOptions());
+        this.exasolFactory = new ExasolObjectFactory(this.exasolConnection, builder.build());
+        this.hanaFactory = new HanaObjectFactory(this.hanaConnection);
+        final ExasolSchema exasolSchema = this.exasolFactory.createSchema(SCHEMA_EXASOL);
+
+        this.adapterScript = createAdapterScript(exasolSchema);
+        final String connectionString = "jdbc:sap://" + this.exasolContainer.getHostIp() + ":"
+                + this.hanaContainer.getSystemPort() + "/";
+        this.connectionDefinition = this.exasolFactory.createConnectionDefinition("HANA_CONNECTION", connectionString,
+                this.hanaContainer.getUsername(), this.hanaContainer.getPassword());
+    }
+
+    public static IntegrationTestSetup start() {
+        final HanaContainer<?> hana = new HanaContainer<>("saplabs/hanaexpress:2.00.061.00.20220519.1").withReuse(true);
+        final ExasolContainer<?> exasol = new ExasolContainer<>("7.1.17")
+                .withRequiredServices(ExasolService.BUCKETFS, ExasolService.UDF).withReuse(true);
+        hana.start();
+        exasol.start();
+        final Bucket bucket = exasol.getDefaultBucket();
+        uploadDriverToBucket(bucket);
+        uploadVsJarToBucket(bucket);
+        try {
+            return new IntegrationTestSetup(hana, exasol);
+        } catch (NoDriverFoundException | SQLException exception) {
+            throw new IllegalStateException("Failed to start integration test setup", exception);
+        }
+    }
+
+    private static void uploadDriverToBucket(final Bucket bucket) {
```

Comment:
Use `ExasolContainer`'s `DriverManager`

Reply:
I didn't know that. Smart!

Reply:
Reverted to old code as there is an issue in the driver manager: https://github.com/exasol/exasol-testcontainers/issues/219